tests/perf_echomail_analysis.php

```
@@ -0,0 +1,849 @@
+if ($currentT['avg'] > 10) {
+    printf("saves ~%s\n", fmtMs($currentT['avg']));
+} else {
+    echo "minimal at current scale\n";
+}
+echo "    Fix 2 (cached last_post_* replaces DISTINCT ON): ";
+if ($t['avg'] > 50) {
+    printf("saves ~%s\n", fmtMs($t['avg']));
+} else {
+    echo "currently fast, will worsen as message base grows\n";
+}
+if ($hasCachedCols && isset($echolistSpeedup)) {
```

PHPStan (2.1.51)

[error] Variable $echolistSpeedup in isset() always exists and is not nullable.


(isset.variable)